libs/sdk-ui-theme-provider/src/complementaryPalette.ts

```diff
@@ -0,0 +1,54 @@
+// (C) 2021 GoodData Corporation
+import { mix } from "polished";
+import { IThemeComplementaryPalette } from "@gooddata/sdk-backend-spi";
+
+const COMPLEMENTARY_PALETTE_PREFIX = "shade";
+
+type IComplementaryPaletteKey =
+    | "shade0"
+    | "shade1"
+    | "shade2"
+    | "shade3"
+    | "shade4"
+    | "shade5"
+    | "shade6"
+    | "shade7"
+    | "shade8"
+    | "shade9";
```

Comment:
i wonder can you use keyof here and have the keys derived from the type?

Reply:
Absolutely.